src/tasks/copy-certs.ts

```diff
@@ -0,0 +1,78 @@
+import { V1Secret, V1SecretList } from '@kubernetes/client-node'
+import { env } from '../common/envalid'
+import { getApiClient } from '../common/k8s'
+import { getFilename, OtomiDebugger, terminal } from '../common/utils'
+
+const cmdName = getFilename(__filename)
+const debug: OtomiDebugger = terminal(cmdName)
+
+let targetNamespace = 'istio-system'
+const client = getApiClient()
+
+const getIstioTlsSecretNames = async (): Promise<string[]> => {
+  const istioTlsSecretsRes = await client.listNamespacedSecret(
+    'istio-system',
+    undefined,
+    undefined,
+    undefined,
+    'type=kubernetes.io/tls',
+  )
+  const { body: istioTlsSecrets }: { body: V1SecretList } = istioTlsSecretsRes
+
+  const istioTlsSecretNames = istioTlsSecrets.items.map((s: V1Secret) => s.metadata!.name!)
+  debug.debug(`Found the following TLS secrets in the ${targetNamespace}: ${istioTlsSecretNames}`)
+  return istioTlsSecretNames
+}
+
+const copyTeamTlsSecrets = async (teamId, istioTlsSecretNames): Promise<void> => {
+  debug.info(`Copying TLS secrets from team-${teamId} to istio-system namespace`)
+  const namespace = `team-${teamId}`
+  const getIstioSecretName = (metadata) => `copy-${teamId}-${metadata.name}`
+  const teamTlsSecretsRes = await client.listNamespacedSecret(
+    namespace,
+    undefined,
+    undefined,
+    undefined,
+    'type=kubernetes.io/tls',
+  )
+  const {
+    body: { items: teamTlsSecrets },
+  } = teamTlsSecretsRes
+  try {
+    await Promise.all(
+      teamTlsSecrets
+        .filter(({ metadata }) => !istioTlsSecretNames.includes(getIstioSecretName(metadata)))
+        .map(({ metadata, data }) => {
+          const name = getIstioSecretName(metadata)
+          debug.info(`Creating TLS secret "${targetNamespace}/${name}"`)
+          const newSecret: V1Secret = {
+            ...new V1Secret(),
+            metadata: {
+              namespace: targetNamespace,
+              name,
+              annotations: {
+                'app.kubernetes.io/managed-by': 'otomi',
+                'log.otomi.io/copied-from-namespace': teamId,
+              },
+            },
+            type: 'kubernetes.io/tls',
+            data,
+          }
+          return client.createNamespacedSecret(targetNamespace, newSecret)
+        }),
+    )
+    debug.info(`Finished copying TLS secrets from team-${teamId}`)
+  } catch (e) {
+    throw new Error(`One or more errors occurred copying secrets: ${JSON.stringify(e)}`)
+  }
+}
+
+export default async (values: Record<string, any>): Promise<void> => {
+  if (values.otomi.hasCloudLB) targetNamespace = 'ingress'
+  const istioTlsSecretNames = await getIstioTlsSecretNames()
+  await Promise.all(
+    env.TEAM_IDS.map((teamId) => {
```

Comment:
If there is access to the values then why TEAM_IDS are provided by env?

Reply:
I disagree with all of your observations and conclusions. 

Spaghetti code? Show me.

About env vars needed for jobs: workloads run on the cluster, and have to be provided with env. Or do you think the values are magically available? That is only so in the Drone pipeline.

Please give me _one_ good reason with concrete arguments that this code is introducing bad practice.

Reply:
I might have made the mistake to want to access those values but those need to be provided by the job as well.

Reply:
I can kinda understand your argument that whenever there is an observation one needs to be specific what it is. So I think you can definitely ask what "exactly" constitutes as spaghetti code. However, I agree with Jeho's observation that rewriting our previous approach can and has, in other instances, lead to ambiguous and confusing code. 

Ie., it can sometimes be easier to say "the codebase is buggy", instead of making one separatie issue for each of the e.g. 1000 bugs that exist in the codebase.

Reply:
Just don't mix CLI with tasks ;)
Write simple file that can be executed without whole CLI entry point and provide interface that serves only that tasks. It is not that hard and will keep code clean

Reply:
> Just don't mix CLI with tasks ;)
> 

What do you mean mix? I have not heard any argument against it that makes enough sense to NOT do everything with the CLI. Everything should be done via the CLI! Please make any good argument against it. I can make plenty that are for doing this with the CLI.

Reply:
> However, I agree with Jeho's observation that rewriting our previous approach can and has, in other instances, lead to ambiguous and confusing code. 
>

What previous approach? This is just introduced by me, and, to use your own words, can you also please be specific with your arguments then?
 
> Ie., it can sometimes be easier to say "the codebase is buggy", instead of making one separatie issue for each of the e.g. 1000 bugs that exist in the codebase. 

Not helpful in these discussions, no

Reply:
Just stay consistent and put tasks in otomi/tasks? Or put them all in the CLI. Just pick something. 

Also, I wonder why you want a code review if you don't want to listen to anyone.

Reply:
We are moving towards developing tasks in core. We do that progressively. Side by side until it is moved.

Reply:
But we do not agree with this implementation.
Consider migrating tasks like: harbor, keycloak etc.
Would you add all these env variables as CLI entry point? 
Are they going to be mandatory for all CLI commands or options?

In my opinion tasks should define its own interface and entry point.

Reply:
> But we do not agree with this implementation.
We? You? I am the lead architect and want this.

> Consider migrating tasks like: harbor, keycloak etc.
Progressively, as always. No major rework in one go necessary. We still have old bash stuff that is not ported after a year and that is also fine.

> Would you add all these env variables as CLI entry point? Are they going to be mandatory for all CLI commands or options?

the envalid setup is modified accordingly so please have another look

> 
> In my opinion tasks should define its own interface and entry point.

Interface? This is 12 factor design so that is env. The interface in in the partial plucking of the `envalid.ts/taskEnvSpec`